internal/api/keppel/accounts.go

```diff
@@ -394,6 +415,10 @@ func (a *API) handlePutAccount(w http.ResponseWriter, r *http.Request) {
 			http.Error(w, err.Error(), http.StatusUnprocessableEntity)
 			return
 		}
+		if rbacPolicies[idx].CanFirstPullAnonymously && req.Account.ReplicationPolicy == nil {
```

Comment:
This only checks for replication, not for external replication:
```suggestion
		rp := req.Account.ReplicationPolicy
		if rbacPolicies[idx].CanFirstPullAnonymously && (rp == nil || rp.ExternalPeer.URL == "") {
```